src/Serilog.Sinks.File/FileLoggerConfigurationExtensions.cs

```
@@ -309,11 +315,16 @@ public static class FileLoggerConfigurationExtensions
 #pragma warning disable 618
                     if (shared)
                     {
+                        if (hooks != null)
+                        {
+                            SelfLog.WriteLine("Unable to use output stream wrapper - these are not supported for shared log files");
```

maybe having a message being a bit more detailed could help track down who wrote this `SelfLog` message when investigating issues ? 

I think it would be good if the message at least mentioned `FileLifecycleHooks` to get an idea of what the user is doing wrong.

---

Also, if I remember the guidelines of `Serilog` properly, we consider that it's *OK* to throw / crash during configuration of the pipeline when it is a "developer error" (but should not throw later on).

see https://github.com/serilog/serilog/wiki/Reliability#configuration for more.
Probably @nblumhardt has a clearer/more opinionated view on the philosophy :)

---

> Also, if I remember the guidelines of `Serilog` properly, we consider that it's _OK_ to throw / crash during configuration of the pipeline when it is a "developer error" (but should not throw later on).
> 
> see https://github.com/serilog/serilog/wiki/Reliability#configuration for more.
> Probably @nblumhardt has a clearer/more opinionated view on the philosophy :)

I purposely didn't throw because I remembered Serilog's "no throw" philosophy :)   Easily changed if throwing is allowed at startup tho

---

> 
> 
> maybe having a message being a bit more detailed could help track down who wrote this `SelfLog` message when investigating issues ?
> 
> I think it would be good if the message at least mentioned `FileLifecycleHooks` to get an idea of what the user is doing wrong.

Very sensible - I just pushed e604418ccaad879556a01b33aa6d34b6dac1c759 for this

---

Great point @tsimbalar - this is one of the cases (incorrect arguments) where I think we usually throw. (E.g. this could be an `ArgumentException` like the ones above in https://github.com/serilog/serilog-sinks-file/blob/dev/src/Serilog.Sinks.File/FileLoggerConfigurationExtensions.cs#L292 ).

---

Just pushed e310ab9bd2ab7648d4629b994714240be53a24d6 to throw in this case, instead of just logging. Also inlcluded a test for good measure.

---

Awesome 👍 - I will check out the overload situation later today, and with some luck, push this all through to dev! :-)